pkgs/development/python-modules/doit/default.nix

```diff
@@ -3,40 +3,53 @@
 , fetchPypi
 , buildPythonPackage
 , isPy3k
-, mock
-, pytestCheckHook
 , cloudpickle
 , pyinotify
 , macfsevents
 , toml
+, importlib-metadata
+, callPackage
 }:
 
 buildPythonPackage rec {
   pname = "doit";
-  version = "0.35.0";
+  version = "0.36.0";
 
   disabled = !isPy3k;
 
   src = fetchPypi {
     inherit pname version;
-    sha256 = "sha256-cVoyLIdMTLhiOU46DWn/MlcrUln1cDb7/cEFPEwB00g=";
+    sha256 = "sha256-cdB8zJUUyyL+WdmJmVd2ZeqrV+FvZE0EM2rgtLriNLw=";
   };
 
+  outputs = [
+    "out"
+    "testsout"
+  ];
+
   propagatedBuildInputs = [
     cloudpickle
     toml
   ] ++ lib.optional stdenv.isLinux pyinotify
     ++ lib.optional stdenv.isDarwin macfsevents;
 
-  # hangs on darwin
-  doCheck = !stdenv.isDarwin;
+  buildInputs = [
+    importlib-metadata
+  ];
 
-  checkInputs = [ mock pytestCheckHook ];
+  postInstall = ''
+    mkdir $testsout
+    cp -R . $testsout/
+  '';
 
-  disabledTests = [
-    # depends on doit-py, which has a circular dependency on doit
-    "test___main__.py"
-  ];
+  pythonImportsCheck = [ "doit" ];
+
+  # check in passthru.tests.pytest to escape infinite recursion with doit-py
+  doCheck = false;
+
+  passthru.tests = {
+    pytest = callPackage ./tests.nix { };
```

Comment:
Suggest using the self-override pattern discussed for `bsuite` rather than the slightly odd dummy-package approach.